hazelcast-client-new/src/main/java/com/hazelcast/client/cache/impl/AbstractClientInternalCacheProxy.java

```diff
@@ -409,11 +556,35 @@ protected ClientInvocationFuture invoke(ClientMessage req, Data keyData, int com
         } catch (Exception e) {
             throw ExceptionUtil.rethrow(e);
         }
-        return new ClientDelegatingFuture<Boolean>(future, clientContext.getSerializationService(),
-                putIfAbsentResponseDecoder);
+        ClientDelegatingFuture delegatingFuture =
+                new ClientDelegatingFuture<Boolean>(future, clientContext.getSerializationService(),
+                                                    putIfAbsentResponseDecoder);
+        if (async && statisticsEnabled) {
+            delegatingFuture.andThen(new ExecutionCallback<Object>() {
```

Comment:
@serkan-ozal  I did not understand. As far as I know, they are running on same executor, aren't they ? I mean, where is it running on when we choose to make the call inside the `putIfAbsentAsyncInternal ` method?